question-service/src/routes/questionRoutes.ts

```diff
@@ -0,0 +1,59 @@
+import { Request, Response } from "express";
+import express from "express";
+import Question from "../models/questionModel";
+
+const router = express.Router();
+
+
+// Add a question
+router.post("/addquestion", async (req: Request, res: Response) => {
+  try {
+    const { title, description, difficulty, topics } = req.body;
+    
```

Comment:
```suggestion
    if (!title || typeof title !== 'string' || title.trim() === '') {
        return res.status(400).json({ message: "Title is required and must be a non-empty string." });
    }
    if (!description || typeof description !== 'string' || description.trim() === '') {
        return res.status(400).json({ message: "Description is required and must be a non-empty string." });
    }
    const allowedDifficulties = ["easy", "medium", "hard"];
    if (!difficulty || !allowedDifficulties.includes(difficulty)) {
        return res.status(400).json({ message: "Difficulty is required and must be one of 'easy', 'medium', or 'hard'." });
    }
    if (!topics || !Array.isArray(topics) || !topics.every(t => typeof t === 'string')) {
        return res.status(400).json({ message: "Topics must be an array of strings." });
    }
```

I think some validations for the title, description, difficulties and topics should be added in the post section for better clarity of errors, improving user experience (and dev debugging) and catching any errors before the database throws any errors.